pkg/smokescreen/smokescreen_test.go

```diff
@@ -1069,7 +1198,7 @@ func proxyServer(conf *Config) *httptest.Server {
 	return httptest.NewServer(proxy)
 }
 
-func proxyClient(proxy string) (*http.Client, error) {
+func proxyClient(proxy string, proxyConnectHeaders http.Header) (*http.Client, error) {
```

Comment:
nit: Instead of reusing this function and having to update all of the existing call sites to pass in a nil `proxyConnectHeaders` my preference would be to create a new `proxyClientWithConnectHeaders` function that wraps the original `proxyClient` function.

Reply:
done!